### PR TITLE
4.1.1 Final

### PR DESCRIPTION
**Pull request by mikebell90**

### CODE REVIEW

File: clientfactory-resteasy-apache/pom.xml [hunk 1]
```diff
@@ -65,11 +73,12 @@
       <artifactId>httpclient</artifactId>
     </dependency>
 
+    <!--
```
[the remark this answers is not in the document]
Comment (adamzr) [reply clientfactory-resteasy-apache/pom.xml]: Can we remove?

File: clientfactory-resteasy-apache/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java [hunk 2]
```diff
@@ -133,12 +135,13 @@ private void configureAuthenticationIfNeeded(String clientName, ResteasyClientBu
...
 
-    private void configureThreadPool(String clientName, ResteasyClientBuilder clientBuilder, JaxRsClientConfig config) {
+    private ExecutorService configureThreadPool(String clientName, JaxRsClientConfig config) {
         final ExecutorService executor = new ThreadPoolExecutor(1, calculateThreads(config.getExecutorThreads()), 1, TimeUnit.HOURS,
                 requestQueue(config.getAsyncQueueLimit()),
                 new ThreadFactoryBuilder().setNameFormat(clientName + "-worker-%s").build(),
                 new ThreadPoolExecutor.AbortPolicy());
-        clientBuilder.asyncExecutor(executor, true);
+        return executor;
+       // clientBuilder.asyncExecutor(executor, true);
```
Comment (adamzr) [inline clientfactory-resteasy-apache/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: Can remove

File: clientfactory-resteasy/src/main/java/org/jboss/resteasy/client/jaxrs/JettyResteasyClientBuilder.java [hunk 3]
```diff
@@ -1,15 +1,33 @@
 package org.jboss.resteasy.client.jaxrs;
 
+import java.security.KeyStore;
+import java.util.ArrayList;
+import java.util.Arrays;
```
[the remark this answers is not in the document]
Comment (adamzr) [reply clientfactory-resteasy/src/main/java/org/jboss/resteasy/client/jaxrs/JettyResteasyClientBuilder.java]: 👍 for revisiting. I'd like to dramatically customizations as much as possible.

File: clientfactory-resteasy-apache/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikebell90) [reply clientfactory-resteasy-apache/pom.xml]: Keeping for now a sa note for when we triage between comparing
resteasy-bom, parent pom and these impls

File: clientfactory-resteasy-apache/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java [hunk 4]
```diff
@@ -60,10 +61,13 @@ public JaxRsClientFactoryImpl(ApplicationContext ctx) {
...
 
     public ClientBuilder newBuilder(String clientName, JaxRsClientConfig config) {
-        final ResteasyClientBuilder builder = new ResteasyClientBuilder();
+        final ExecutorService executorService = configureThreadPool(clientName, config);
+        final ResteasyClientBuilderImpl builder = new ResteasyClientBuilderImpl();
+        builder.asyncExecutor(executorService, true);
         configureHttpEngine(clientName, builder, config);
         configureAuthenticationIfNeeded(clientName, builder, config);
-        configureThreadPool(clientName, builder, config);
+        final ExecutorService executor = configureThreadPool(clientName, config);
```
[the remark this answers is not in the document]
Comment (scottjohnson) [reply clientfactory-resteasy-apache/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: Agreed.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,14 @@
...
 ===================
+3.0.0
+----
+* Resteasy 4.0.0 final (non beta) out at last.
+* This has led to significant maven restructuring, due to splits between
+api and implementation packages, and renaming of several packages. The parent
+will do it's best to arbitrate this, but common changes include
+
+(Note the changes)
+* Jersey updated to 2.27. We do not formally support Jersey (though a couple projects use it).
```
[the remark this answers is not in the document]
Comment (mikebell90) [reply CHANGELOG.md]: They have mostly

File: clientfactory-resteasy-apache/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (mikebell90) [reply clientfactory-resteasy-apache/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: nope

[the remark this answers is not in the document]
Comment (mikebell90) [reply clientfactory-resteasy-apache/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: Please recheck

File: clientfactory-resteasy/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java [hunk 6]
```diff
@@ -67,38 +71,86 @@ public JaxRsClientFactoryImpl(ApplicationContext ctx) {
...
+                hc.setUserAgentField(new HttpField(HttpHeader.USER_AGENT, config.getUserAgent()));
+            });
+        }
+        if (config.isCookieHandlingEnabled()) {
+            httpClientCustomizers.add(hc -> hc.setCookieStore(new HttpCookieStore()));
+        }
+        if(StringUtils.isNotBlank(config.getProxyHost()) && config.getProxyPort() != 0) {
+            httpClientCustomizers.add(hc -> hc.getProxyConfiguration().getProxies().add(new HttpProxy(config.getProxyHost(), config.getProxyPort())));
+        }
+        httpClientCustomizers.add(hc -> hc.setMaxConnectionsPerDestination(Math.max(64, config.getHttpClientDefaultMaxPerRoute())));
```
[the remark this answers is not in the document]
Comment (mikebell90) [reply clientfactory-resteasy/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: There is in fact a comment to that effect in next class

File: clientfactory-resteasy-apache/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (adamzr) [reply clientfactory-resteasy-apache/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: What am I missing?

**Pull request merged**